### DISCUSSION TITLE
Request for Japanese Translations in test-prof Gem Documentation

### DISCUSSION THREAD

**Discussion by nodematerial** [test-prof/test-prof#288] [Ideas]
Hello, I hope this message finds you well. I'm a  enthusiast of the test-prof Gem, currently residing in Japan.

I humbly propose the inclusion of Japanese translations within the documentation.

Within my community, the utilization of this Gem is widespread! However, many of us who are not proficient in English often encounter challenges comprehending the official documentation, thus limiting their ability to use diverse functionalities of this Gem.

Consequently, it would be immensely beneficial and delightful if we were able to access Japanese translations in addition to the Russian and Chinese translations within the official documentation.

While I'm uncertain about the logistics of hosting the official documentation and providing translations, I would be more than willing to contribute to the effort, whether through manual translation or any other means.

Warm regards.

![image](https://github.com/test-prof/test-prof/assets/69585273/d889c865-d49d-49a9-ba58-893823388d76)

**Comment by palkan** [comment]
Hey @nodematerial,

I'm happy to accept your proposal 🙂 Let's do that.

Actually, we started to work on the Japanese translation a while ago but didn't manage to achieve a lot due to lack of time.

Here is the repo (just made it public): https://github.com/test-prof/docs-jp

The flow is as follows:

- Copy a documentation page from https://github.com/test-prof/test-prof
- Put it under the path in the JP repo
- Translate it!

The process is managed via PRs, as usual.

The way we did it for Ru/Cn translations, we copied the whole English folder and gradually updated docs one by one. This way, we can enable JP translation at the website as soon as possible, but still some pages would be in English (added a note at the top saying the "Translation is in progress; you can contribute here" would be good, too).

If you'd like to lead the project I'd be happy to add you as a collaborator, so you can approve and merge PRs and push to master 😁

We don't have any synchronization mechanism yet, so keeping docs up-to-date is a manual work for now (sometimes, for significant changes, I open issues in the localization repos).

Thank you!

**Reply by nodematerial** [reply]
Thanks for your reply! 
I would like to contribute to JP repo, so I forked the repository and begun to translate it.
I'll report back on the situation after a while.

**Reply by nodematerial** [reply]
Hi @palkan

I sent the first pull request to `test-prof/docs-jp` https://github.com/test-prof/docs-jp/pull/1
Then, I would like you to review my pull request.

And, I want to be a collaborator if I can be.

**Reply by palkan** [reply]
Done!

Also, renamed the repo to `docs-ja` (looks like "ja" is a correct abbreviation for Japanese). Will add a locale switch to the website soon.

**Reply by palkan** [reply]
Okay, here we go: https://test-prof.evilmartians.io/ja/

**Reply by nodematerial** [reply]
Thank you for your review!
